packages/db/src/core/integration/vite-plugin-db.ts

```diff
@@ -124,12 +124,14 @@ export function getLocalVirtualModContents({
 	tables: DBTables;
 	root: URL;
 }) {
+	const { ASTRO_DATABASE_FILE } = getAstroEnv();
+	const envDbUrl = ASTRO_DATABASE_FILE ? `'${ASTRO_DATABASE_FILE}'` : undefined;
```

Comment:
I don't think you need to stringify the string, because it's already a string. We already to such here:

https://github.com/withastro/astro/blob/ff522b96a01391a29b44f820dfcc2a2176d871e7/packages/db/src/core/integration/vite-plugin-db.ts#L211-L212

Reply:
It will be
```ts
const dbUrl = normalizeDatabaseUrl(file://local.db, ${JSON.stringify(dbUrl)});
```
without stringification.

But I just realized I could have used JSON.stringify, so I'll update.

Reply:
I see. The failing test is definitely weird though, I wonder what's causing the failure

Reply:
I think the test passed before because ASTRO_DATABASE_URL was not passed to the plugin at all.

And this line of the test generates a malformed URL on Windows:
https://github.com/withastro/astro/blob/ff522b96a01391a29b44f820dfcc2a2176d871e7/packages/db/test/local-prod.test.js#L39

Reply:
Hopefully someone with Windows machine can fix that.

Reply:
Wait it passed now 💀

Reply:
I'm not sure what magic JSON.stringify does here.